xds/src/main/java/io/grpc/xds/LookasideLb.java

```diff
@@ -248,6 +194,208 @@ private static ManagedChannel initLbChannel(
     return channel;
   }
 
+  /**
+   * A LoadBalancerProvider that provides a load balancer with a fixed XdsConfig.
+   */
+  private final class FixedXdsConfigBalancerProvider extends LoadBalancerProvider {
+    final XdsConfig xdsConfig;
+    final XdsConfig oldXdsConfig;
+    final EndpointWatcher oldEndpointWatcher;
+
+    FixedXdsConfigBalancerProvider(XdsConfig xdsConfig) {
+      this.xdsConfig = xdsConfig;
+      oldXdsConfig = LookasideLb.this.xdsConfig;
+      oldEndpointWatcher = endpointWatcher;
+    }
+
+    @Override
+    public boolean isAvailable() {
+      return true;
+    }
+
+    @Override
+    public int getPriority() {
+      return 5;
+    }
+
+    // A synthetic policy name identified by xds config.
+    @Override
+    public String getPolicyName() {
+      return "xds_policy__balancer_name_" + xdsConfig.balancerName
+          + "__childPolicy_" + xdsConfig.childPolicy
+          + "__fallbackPolicy_" + xdsConfig.fallbackPolicy
+          + "__edsServiceName_" + xdsConfig.edsServiceName
+          + "__lrsServerName_" + xdsConfig.lrsServerName;
+    }
+
+    @Override
+    public LoadBalancer newLoadBalancer(Helper helper) {
+      return new FixedXdsConfigBalancer(helper);
+    }
+
+    final class FixedXdsConfigBalancer extends LoadBalancer {
+      final Helper helper;
+
+      // All fields become non-null once handleResolvedAddresses() successfully.
+      // All fields are assigned at most once.
+      @Nullable
+      ObjectPool<XdsClient> xdsClientRef;
+      @Nullable
+      XdsClient xdsClient;
+      @Nullable
+      LocalityStore localityStore;
+      @Nullable
+      LoadReportClient lrsClient;
+
+      FixedXdsConfigBalancer(Helper helper) {
+        this.helper = helper;
+      }
+
+      @Override
+      public void handleNameResolutionError(Status error) {}
+
+      @Override
+      public boolean canHandleEmptyAddressListFromNameResolution() {
+        return true;
+      }
+
+      @Override
+      public void handleResolvedAddresses(ResolvedAddresses resolvedAddresses) {
+        if (xdsClientRef != null) {
+          return;
+        }
+
+        LoadStatsStore loadStatsStore = new LoadStatsStoreImpl();
+        localityStore = localityStoreFactory.newLocalityStore(
+            helper, lbRegistry, loadStatsStore);
+        // TODO(zdapeng): Use XdsClient to do Lrs directly.
+        final LoadReportCallback lrsCallback =
+            new LoadReportCallback() {
+              @Override
+              public void onReportResponse(long reportIntervalNano) {
+                localityStore.updateOobMetricsReportInterval(reportIntervalNano);
+              }
+            };
+
+        // There are three usecases:
+        // 1. The EDS-only legacy usecase that requires balancerName from xds config.
```

Comment:
Delete the case for using balancer name from xds config. It is ALREADY deprecated. Internal test will have the bootstrap file for you. It contains serverUri same as balancerName we previously got from service config. Internal tests are waiting for us to deprecate this usage.

Reply:
This is the only case interop-tested. Once the bootstrap is interop-tested and working, we can delete the deprecated one.

Reply:
What do you mean "bootstrap is interop-tested"? Reading bootstrap is adopted in internal tests. The test infrastructure supports generating a bootstrap file for us. It's already there.

Reply:
It's already there but we did not test it.

Reply:
You mean we have not "tried" that and is worrying about there might be some bugs for the second case here? Alright, you can do that later.